libflux/flux-core/src/db.rs

```diff
@@ -35,6 +36,151 @@ pub enum Error {
     Message(String),
 }
 
+/// Interface for retrieving external flux modules
+pub trait Fluxmod: fmt::Debug + std::panic::RefUnwindSafe {
+    fn get_module(&self, module: &str) -> Option<Vec<(String, Arc<str>)>>;
+}
+
+#[derive(Debug)]
+struct HttpFluxmod {
+    token: String,
+}
+
+impl HttpFluxmod {
+    fn new(token: String) -> Self {
+        HttpFluxmod { token }
+    }
+
+    const BASE: &'static str =
+        "https://twodotoh-dev-markus20221018125005.remocal.influxdev.co/api/v2private/modules";
+
+    #[cfg(all(test, feature = "integration_test"))]
+    fn publish(
+        &self,
+        module: &str,
+        files: Vec<(String, Arc<str>)>,
+        version: &semver::Version,
+    ) -> Result<()> {
+        let mut multipart = multipart::client::lazy::Multipart::new();
+        for (name, contents) in &files {
+            multipart.add_stream("module", contents.as_bytes(), Some(name), None);
+        }
+        let body = multipart
+            .prepare()
+            .map_err(|err| Error::Message(err.to_string()))?;
+
+        let response = ureq::post(&format!("{}/{}/@v/v{}.zip", Self::BASE, module, version))
```

Comment:
Heads up, we expect the `.zip` extention to be removed from the API route since we are not posting zip files. We will use the `Content-Type` header to indicate how the data is being uploaded. Obviously this works now, but when we change the server we will need to update this line.